src/impl/apple/keychain_helper.cpp

```diff
@@ -119,10 +119,8 @@ std::vector<char> metadata_realm_encryption_key(bool check_legacy_service)
         service = retainCF(legacy_service);
         check_legacy_service = false;
     } else {
-        // FIXME: Get rid of this manual retain once we fix CFPtr to accept null.
-        CFRetain(bundle_id);
+        // FIXME: Collapse the null check on bundle_id once CFPtr is fixed to accept null pointers.
```

Comment:
I'm not 100% sure what you mean by "collapse the null check" here. We could write this like so, though:

```c++
if (CFStringRef bundle_id = CFBundleGetIdentifier(CFBundleGetMainBundle()))
    service = adoptCF(CFStringCreateWithFormat(NULL, NULL, CFSTR("%@ - Realm Sync Metadata Key"), bundle_id));
else {
    service = retainCF(legacy_service);
    check_legacy_service = false;
}
```

Reply:
We don't actually need to use `retainCF` for the temporary `bundle_id`, so your suggestion addresses the problem and there isn't anything we need to do in the future.